native_client/coqui-stt.h

```diff
@@ -61,6 +61,11 @@ typedef struct Metadata {
   const CandidateTranscript* const transcripts;
   /** Size of the transcripts array */
   const unsigned int num_transcripts;
+  /** Logits and information to decode them **/
+  const char* const alphabet;
```

Comment:
I guess this is just for the first pass but the output data structures should have decoded characters rather than including an Alphabet serialization in it.

Reply:
Yeah, it's the first pass. I didn't want to do anything fancy because I'd probably have to change it anway. What is the suggestion for a C-compatible structure for encoding a matrix of double X character? Should I make some structs, something like:

```c
struct Symbol {
    char *symbol;
}
struct Timestep {
    Symbols **symbols;
    double **softmax;
} 
```
Then just include `Timestep **timesteps` ?

The issue is that because of UTF-8 stuff, we can't assume that `char*` is a single symbol. So we have to wrap it in some more structure.

Reply:
C makes this type of stuff quite verbose. I recommend using a linear array (double*, char*) and having the array dimensions in separate attributes (size of time dimension and size of class dimension), and then document if the memory layout is time major or class major. We also need to encode the CTC blank character as a char* somehow (using a special ASCII symbol maybe?).

So maybe something like:

```cpp
struct AcousticModelEmissions {
    int num_symbols;
    char** symbols; // num_symbols long array of NULL-terminated strings
    int num_timesteps;
    double** emissions; // num_timesteps long array, each pointer is a num_symbols long array
}

struct Metadata {
    // ...
    AcousticModelEmissions *emissions;
}
```

Reply:
(This would be a time major layout, you could also encode `emissions` in a symbol major layout. I guess time major is easier to implement as the decoder is already operating in a time major way.

Reply:
Ok, implemented. I made a minor change, which was to use `double* emissions` as I was doing before. I can change it, but could do with a pointer to how to do it (pun intended).